src/main/scala/org/neo4j/spark/Neo4jConfig.scala

```diff
@@ -1,34 +1,41 @@
 package org.neo4j.spark
 
 import org.apache.spark.SparkConf
-import org.neo4j.driver.v1.{Driver, AuthTokens, Config, GraphDatabase}
+import org.neo4j.driver.{AuthToken, AuthTokens, Config, Driver, GraphDatabase, Session, SessionConfig}
 
 /**
  * @author mh
  * @since 02.03.16
  */
-case class Neo4jConfig(val url: String, val user: String = "neo4j", val password: Option[String] = None, encryptionStatus:Boolean) {
+case class Neo4jConfig(val url: String,
+                       val user: String = "neo4j",
+                       val password: Option[String] = None,
+                       val database: Option[String] = None,
+                       val encryptionStatus: Boolean) {
 
-  def boltConfig(): Config = if (encryptionStatus) Config.build().withEncryption().toConfig else Config.build().withoutEncryption().toConfig
+  private def boltConfig(): Config = if (encryptionStatus) Config.builder().withEncryption().build() else Config.builder().withoutEncryption().build()
 
   def driver(config: Neo4jConfig) : Driver = config.password match {
-    case Some(pwd) => GraphDatabase.driver(config.url, AuthTokens.basic(config.user, pwd), boltConfig())
-    case _ => GraphDatabase.driver(config.url, boltConfig())
+    case Some(pwd) => driver(config.url, AuthTokens.basic(config.user, pwd))
+    case _ => driver(config.url, AuthTokens.none())
   }
 
-  def driver() : Driver = driver(this)
+  def driver(): Driver = driver(this)
 
-  def driver(url: String): Driver = GraphDatabase.driver(url, boltConfig())
+  def driver(url: String, authToken: AuthToken): Driver = GraphDatabase.driver(url, authToken, boltConfig())
+
+  def sessionConfig(): SessionConfig = database.map { SessionConfig.forDatabase(_) }.getOrElse(SessionConfig.defaultConfig())
 
 }
 
 object Neo4jConfig {
-  val prefix = "spark.neo4j.bolt."
+  val prefix = "spark.neo4j"
   def apply(sparkConf: SparkConf): Neo4jConfig = {
-    val url = sparkConf.get(prefix + "url", "bolt://localhost")
-    val user = sparkConf.get(prefix + "user", "neo4j")
-    val password: Option[String] = sparkConf.getOption(prefix + "password")
-    val encryptionStatus : Boolean = sparkConf.getBoolean(prefix + "encryption.status", defaultValue = false)
-    Neo4jConfig(url, user, password,encryptionStatus)
+    val url = sparkConf.get(s"$prefix.bolt.url", "bolt://localhost")
+    val user = sparkConf.get(s"$prefix.bolt.user", "neo4j")
+    val password: Option[String] = sparkConf.getOption(s"$prefix.bolt.password")
+    val database: Option[String] = sparkConf.getOption(s"$prefix.database")
```

Comment:
why no bolt here ? seems inconsistent?

Reply:
Because it's not bolt strictly related to I used `spark.neo4j.database`
Do you prefer `spark.neo4j.bolt.database`?

Reply:
either we use bolt everywhere or we drop it everywhere (then we have to update docs + readme)
user and password are also not related to bolt :)

Reply:
Driver changes in 4.0 have sadly made this one complicated.   bolt will continue to work, but it's deprecated, and implies the wrong thing for 3.5 cluster.   Neo4j is new hotness -- but *does not* work with 3.5.

My vote would be to make it `spark.neo4j.database` and use "neo4j" throughout to be supportive of the overall messaging for Neo4j.  But 100% we're going to need some special instructions for migration of the old driver, and an explanation of all of this bolt, bolt+routing, and neo4j craziness.

Reply:
@jexp you're right I'm removing bolt from everywhere and updating the docs.